clients/drcachesim/tests/window_test.cpp

```diff
@@ -0,0 +1,185 @@
+{
+#ifdef UNIX
+    return setenv(var, value, 1 /*override*/) == 0;
+#else
+    return SetEnvironmentVariable(var, value) == TRUE;
+#endif
+}
+
+int
+fib_with_sleep(int n)
+{
+    dr_fprintf(STDERR, "Calculating fibonacci of: %d\n", n);
+    std::this_thread::sleep_for(std::chrono::seconds(SECONDS_TO_SLEEP));
+    if (n <= 1)
+        return 1;
+    return fib_with_sleep(n - 1) + fib_with_sleep(n - 2);
+}
+
+static std::string
+post_process()
+{
+    const char *raw_dir;
+    drmemtrace_status_t mem_res = drmemtrace_get_output_path(&raw_dir);
+    assert(mem_res == DRMEMTRACE_SUCCESS);
+    std::string outdir = std::string(raw_dir) + DIRSEP + "trace";
+    void *dr_context = dr_standalone_init();
+    {
+        raw2trace_directory_t dir;
+        if (!dr_create_dir(outdir.c_str())) {
+            std::cerr << "Failed to create output dir";
+            assert(false);
+        }
+        std::string dir_err = dir.initialize(raw_dir, outdir);
+        assert(dir_err.empty());
+        raw2trace_t raw2trace(dir.modfile_bytes_, dir.in_files_, dir.out_files_,
+                              dir.out_archives_, dir.encoding_file_,
+                              dir.serial_schedule_file_, dir.cpu_schedule_file_,
+                              dr_context,
+                              0
+#ifdef WINDOWS
+                              /* TODO i#3983: Creating threads in standalone mode
+                               * causes problems.  We disable the pool for now.
+                               */
+                              ,
+                              0
+#endif
+        );
+        std::string error = raw2trace.do_conversion();
+        if (!error.empty()) {
+            std::cerr << "raw2trace failed: " << error << "\n";
+            assert(false);
+        }
+    }
+    dr_standalone_exit();
+    return outdir;
+}
+
+static std::string
+gather_trace()
+{
+    std::string dr_ops(
+        "-stderr_mask 0xc -client_lib ';;-offline -offline -trace_after_instrs 1000 "
+        "-trace_for_instrs 2500 -retrace_every_instrs 1000");
+    if (!my_setenv("DYNAMORIO_OPTIONS", dr_ops.c_str()))
+        std::cerr << "failed to set env var!\n";
+    dr_app_setup();
+    assert(!dr_app_running_under_dynamorio());
+    dr_app_start();
+    assert(dr_app_running_under_dynamorio());
+    fib_with_sleep(6);
+    dr_app_stop_and_cleanup();
+    assert(!dr_app_running_under_dynamorio());
+    return post_process();
+}
+
+int
+test_main(int argc, const char *argv[])
+{
+    std::string dir = gather_trace();
+    void *dr_context = dr_standalone_init();
+    scheduler_t scheduler;
+    std::vector<scheduler_t::input_workload_t> sched_inputs;
+    sched_inputs.emplace_back(dir);
+    if (scheduler.init(sched_inputs, 1, scheduler_t::make_scheduler_serial_options()) !=
+        scheduler_t::STATUS_SUCCESS) {
+        std::cerr << "Failed to initialize scheduler " << scheduler.get_error_string()
+                  << "\n";
```

Comment:
This seems like a fatal error that should not continue.

Reply:
Done.